extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -43,21 +44,31 @@ export class AmpExperiment extends AMP.BaseElement {
 
     const config = this.getConfig_();
     const results = Object.create(null);
+    const override = this.getVariantOverride_();
     const variants = Object.keys(config).map(experimentName => {
+      if (override[experimentName]) {
+        results[experimentName] = override[experimentName];
```

Comment:
This is currently not a valid example since amp-img doesn't allow URL substitution.

However, I do agree we can improve the code to avoid any risk in future. Probably moving the logic into `variant.js` would be the ideal way.